Sources/MacroToolkit/Type.swift

```diff
@@ -174,6 +169,199 @@ public enum `Type`: TypeProtocol, SyntaxExpressibleByStringInterpolation {
+            optionalTypeSyntax.wrappedType = TypeSyntax(normalizedElement._syntax)
+            attributedTypeSyntax?.baseType = TypeSyntax(optionalTypeSyntax)
+            
+//            let identifierSyntax = IdentifierTypeSyntax(
+//                leadingTrivia: optionalTypeSyntax.leadingTrivia,
+//                name: .identifier("Optional"),
+//                genericArgumentClause: .init(
+//                    arguments: .init(arrayLiteral: .init(argument: optionalTypeSyntax.wrappedType))
+//                ),
+//                trailingTrivia: optionalTypeSyntax.leadingTrivia
+//            )
+            
+            var base = "Optional<\(optionalTypeSyntax.wrappedType)>"
+            if let attributedTypeSyntax {
+                base = base.addingAttributes(from: attributedTypeSyntax)
+            }
+            return NormalizedType(stringLiteral: base)
+        case .packExpansion(let type):
+            // Looks like there can only be simple identifiers in pack expansions, with no generics, and therefore we
+            // don't ned to recursively normalize
+            
+            return .packExpansion(.init(type._baseSyntax, attributedSyntax: type._attributedSyntax))
+        case .packReference(let type):
+            // Looks like there can only be simple identifiers in pack references, with no generics, and therefore we
+            // don't ned to recursively normalize
+            
+            return .packReference(.init(type._baseSyntax, attributedSyntax: type._attributedSyntax))
+        case .simple(let type):
+            if type.name == "Void" {
+                return .tuple(.init(.init(elements: [])))
+            }
+            var identifierTypeSyntax: IdentifierTypeSyntax = type._baseSyntax
+            var attributedTypeSyntax: AttributedTypeSyntax? = type._attributedSyntax
+            
+            identifierTypeSyntax.genericArgumentClause = identifierTypeSyntax.genericArgumentClause?.normalized()
+            
+            attributedTypeSyntax?.baseType = TypeSyntax(identifierTypeSyntax)
+            
+            return .simple(.init(identifierTypeSyntax, attributedSyntax: attributedTypeSyntax))
+        case .suppressed(let type):
+            // Normalizing recursively because it may be needed when https://github.com/apple/swift/issues/62906
+            // is fixed. Not handling `_attributedSyntax` because seems like `suppressed` cannot have any attribute
+            var suppressedTypeSyntax: SuppressedTypeSyntax = type._baseSyntax
+            
+            let normalizedConstraint = Type(suppressedTypeSyntax.type).normalized()
+            suppressedTypeSyntax.type = TypeSyntax(normalizedConstraint._syntax)
+                        
+            return .suppressed(.init(suppressedTypeSyntax))
+        case .tuple(let type):
+            if type.elements.count == 1 {
+                let child = type.elements[0]
+                if case .tuple(_) = child {
```

Comment:
I believe you can do `if case .tuple = child` which is a little bit cleaner, but it doesn't really matter either way.

Reply:
Fixed with the updated implementation in b4a470c.